src/numeric/decision_rules.jl

```diff
@@ -1,9 +1,43 @@
-import splines
+abstract AbstractDecisionRule{S,T}
 
-abstract AbstractDecisionRule
-abstract AbtsractCubicSplineDR <: AbstractDecisionRule
+# abstract AbstractCachedDecisionRule{S,T} <: AbstractDecisionRule{S,T}
+# we don't implement that using subtypes anymore
+
+type DecisionRule{S,T} <: AbstractDecisionRule{S,T}
+    grid_exo::S
+    grid_endo::T
+    n_x::Int # number of values (could be a method if there was a nice method name)
+    coefficients::Array{Array{Float64}}
+end
+
+
+type ConstantDecisionRule
+    constants::Vector{Float64}
+end
+
+(dr::ConstantDecisionRule)(x::Vector{Float64}) = dr.constants
+(dr::ConstantDecisionRule)(x::Matrix{Float64}) = repmat(dr.constants',size(x,1),1)
```

Comment:
If you are defining decision rules on two grids, you evaluate dr(x,y). If you only one non-trivial grid, you evaluate on dr(x). In any case, it doesn't matter.